src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -478,6 +491,36 @@ export const Top = styled(Layout.Main)`
   flex-grow: 0;
 `;
 
+type SavedQueryState = AsyncComponent['state'] & {
+  savedQuery?: SavedQuery | null;
+};
+
+class SavedQueryAPI extends AsyncComponent<Props, SavedQueryState> {
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {organization, location} = this.props;
+    if (location.query.id) {
+      return [
+        [
+          'savedQuery',
+          `/organizations/${organization.slug}/discover/saved/${location.query.id}/`,
+        ],
+      ];
+    }
+    return [];
+  }
+
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderBody(): React.ReactNode {
+    const {savedQuery, loading} = this.state;
+    return (
+      <Results {...this.props} savedQuery={savedQuery ?? undefined} loading={loading} />
```

Comment:
Looking at this super quickly (so I haven't checked the behaviour), but since you call `renderBody` from `renderLoading`, the props will be modified after rendering the results for the first time.

`savedQuery: undefined` -> `savedQuery: <data>`. Basically once saved query returns data, your event view will be overridden. So you'll have to consider what happens when you call the url with a param that differs from what is returned in the saved query response (as this could cause visual jank).

I'm not 100% though since again, I haven't read through this thoroughly, so if you have already covered this, then 👏

Reply:
So we'll should be using this [fromSavedQueryOrLocation](https://github.com/getsentry/sentry/pull/25123/files#diff-7f485fdf25bcbaf30657c8d60f9f6760292a3863eb21fd396bd75b2d04778178R395) method to render the view, so the url params will override the saved query data. Is that what you are referring to?